safe-global/safe-wallet-web
# feat: Add StepperFactory and new tx layout

## What it solves

Part of #2021 

## How this PR fixes it

- Adds new global context `ModalProvider` for orchestrating modals in the app
- Adds new folder structure `TxFlow`
- Adds `NewTxMenu`, `ReplaceTxMenu`, `RejectTx`, `TokenTransfer` to the provider
- Adds `TxLayout` component
- Adds `StepperFactory` and uses it for `TokenTransferFlow`

## ToDos

- [x] Close Modal when navigating in the sidebar
- [ ] Resolve TODOs

## How to test it

1. Open the app
2. Press on `New Transaction` in the sidebar
3. Observe the `NewTxMenu` opening as a fullscreen modal
4. Press on `Send Tokens`
5. Observe the new `TokenTransferFlow` opening as a fullscreen modal
6. Navigate to `/queue`
7. Replace an existing transaction
8. Observe the `ReplaceTxMenu` opening as a fullscreen modal
9. Press on either `Send Tokens` or `Reject Transaction`
10. Observe `TokenTransferFlow` or `RejectTx` opening as a fullscreen modal

## Screenshots
<img width="591" alt="Screenshot 2023-05-26 at 11 51 16" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/36c99cc1-aa31-461f-8859-0f2e89f8780c">
<img width="1241" alt="Screenshot 2023-05-26 at 11 51 24" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/f89d6a83-4a56-467e-a13c-916e4256c9bc">
<img width="1145" alt="Screenshot 2023-05-26 at 11 51 35" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/1eb9b485-562f-4072-870a-6317826c5dd3">
<img width="1236" alt="Screenshot 2023-05-26 at 11 51 46" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/06936d2e-f7b0-453c-8c0a-48aae8cc3926">


## Checklist
